app/models/is_reducible.rb

```diff
@@ -1,6 +1,18 @@
 module IsReducible
   extend ActiveSupport::Concern
 
+  def active?
+    raise NotImplementedError.new
```

Comment:
Maybe add some message context to these errors indicating that they are abstract methods, just in case they ever raise so folks can better identify the issue.

Reply:
sure, i can do that